PublishParasoftTestResults/publishtestresults.ts

```diff
@@ -162,64 +163,44 @@ function transformReports(inputReportFiles: string[], index: number)
+                    ruleAnalyzerMap.set(ruleId, analyzerId);
                 } else if (node.attributes.cat == 'GLOBAL') {
-                    ruleIdsWithCatAsGlobal.add(ruleId);
+                    rulesInGlobalCategory.add(ruleId);
                 }
 
-            } else if (bStaticAnalysisResult && bLegacyReport && node.name.endsWith('Viol')) {
+            } else if (bParsingStaticAnalysisResult && bLegacyReport && node.name.endsWith('Viol')) {
                 let ruleId = node.attributes.rule;
-                let analyzerId = mapToAnalyzer(ruleId, node.name);
-                if(!ruleAnalyzerPairs.get(ruleId) && isDtpSettingsValid) {
-                    promisesOfGettingRuleDocUrl.push(getRulesDoc(ruleId, analyzerId));
+                if(!ruleAnalyzerMap.has(ruleId)) {
+                    let analyzerId = mapToAnalyzer(ruleId, node.name);
+                    if (isDtpSettingsValid) {
+                        ruleDocUrlPromises.push(getRuleDoc(ruleId, analyzerId));
+                    }
+                    ruleAnalyzerMap.set(ruleId, analyzerId);
                 }
-                ruleAnalyzerPairs.set(ruleId, analyzerId);
             }
         });
         saxStream.on("closeTag", function (nodeName) {
             if (nodeName == 'CodingStandards') {
-                bStaticAnalysisResult = false;
+                bParsingStaticAnalysisResult = false;
             }
         });
         saxStream.on("error", function (e) {
             tl.warning('Failed to parse ' + report + '. Error was: ' + e.message);
         });
         saxStream.on("end", function() {
-            switch (reportType) {
-                case ReportType.XML_STATIC:
-                    transformToSarif(report);
-                    break;
-                case ReportType.XML_TESTS:
-                    transformToXUnit(report);
-                    break;
-                case ReportType.XML_STATIC_AND_TESTS:
-                    transformToSarif(report);
-                    transformToXUnit(report);
-                    break;
-                case ReportType.XML_SOATEST:
-                    transformToSOATestXUnit(report);
-                    break;
-                case ReportType.XML_STATIC_AND_SOATEST:
-                    transformToSOATestXUnit(report);
-                    transformToSarif(report);
-                    break;
-                case ReportType.XML_XUNIT:
-                    xUnitReports.push(report);
-                    break;
-                default:
-                    tl.warning("Skipping unrecognized report file: " + report);
-            }
-            if (promisesOfGettingRuleDocUrl.length > 0) {
-                postGettingRulesDocs(promisesOfGettingRuleDocUrl).then(() => {
+            if (ruleDocUrlPromises.length > 0) {
+                handleRuleDocsPromises(ruleDocUrlPromises).then(() => {
+                    transformToReport(reportType, report);
```

Comment:
I moved the transformation process inside `then()` because later we'll include the rule document URL after the transformation, which requires all promises to be resolved.